integration/tests/cook/test_basic.py

```diff
@@ -760,3 +762,52 @@ def test_basic_docker_job(self):
         self.assertEqual(resp.status_code, 201)
         job = util.wait_for_job(self.cook_url, job_uuid, 'completed')
         self.assertEqual('success', job['instances'][0]['status'])
+
+    def get_executor(agent_state, instance):
+        for framework in agent_state['frameworks']:
+            for executor in framework['executors']:
+                if executor['id'] == instance['executor_id'][0]:
+                    return executor
```

Comment:
Could this be done with a list comprehension?

Reply:
I couldn't get one working, and my failed ones felt less clear than the nested for loops. If you can come up with one that works, I'm open to it :)